src/TorchSharp/DisposeScope.cs

```diff
@@ -237,17 +237,6 @@ public IReadOnlyList<IDisposable> Attach(IEnumerable<IDisposable> disposables)
 
             var result = new List<IDisposable>();
             foreach (var disposable in disposables) {
-                if (disposable is torch.Tensor tensor) {
-                    if (tensor.OwningDisposeScope == null && !tensor.IsInvalid) {
```

Comment:
Here, it may be worth adding a separate interface for just the 'IsInvalid' rather than expanding on the IDisposable interface, for separation of concerns. Allow Tensor and PackedSequence to implement IValidatable.

Reply:
Not sure I follow, IsInvalid is an existing public method on Tensor and this is casting specifically to use it. And we left it internal on PackedSequence to leave it hidden. Are you saying we want to make both of them public now, under that interface? I can do that just want to make sure I'm certain what you are looking for here as last time we talked about an interface, we agreed against it to keep the API smaller.